proposals/declarative-secrets.md

```diff
@@ -0,0 +1,56 @@
+# Declarative Secrets
+
+## Motiviation
+
+Various projects exist in the wild that require various secrets for them to run properly. Exmaples include:
+
+- https://github.com/lostintangent/codespaces-langchain
+- https://github.com/openai/openai-quickstart-python
+- https://github.com/openai/openai-quickstart-node
+
+Today these projects have to include specific instructions e.g. in their README telling users where to procure what they need to run and then how to set it up as a secret or add it as an environment variable. This currently acts as an impediment to adoption and promotion of dev containers for these projects.
+
+## Goal
+
+Simplify using dev containers for these kinds of projects by supporting secrets as a first-class part of the dev container creation flow.
+
+## Proposal
+
+Add an optional `secrets` to the `devContainer.base.schema.json`. This will be used to declare the secrets needed within the dev container.
```

Comment:
```suggestion
Add an optional `secrets` property to the `devContainer.base.schema.json`. This will be used to declare the secrets needed within the dev container.
```